Sources/Placeholder.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  Placeholder.swift
+//  Kingfisher
+//
+//  Created by Tieme van Veen on 28/08/2017.
+//
+//  Copyright (c) 2017 Wei Wang <onevcat@gmail.com>
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+#if os(macOS)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+public protocol Placeholder {
+    func add(to imageView: ImageView)
+    func remove(from imageView: ImageView)
+}
+
+extension Placeholder where Self: Image {
+    public func add(to imageView: ImageView) { imageView.image = self }
+    public func remove(from imageView: ImageView) { imageView.image = nil }
+}
+
+extension Image: Placeholder {}
+
+extension Placeholder where Self: View {
```

Comment:
I still don't get why this is better than `extension Placeholder : View {...}`. Could you explain it if you have the time for it. Nevermind otherwise :)

Reply:
@teameh 

By `extension View: Placeholder {...}`, you will make ALL subclasses of UIView to be a `Placeholder`. It could be too wide. While by using `extension Placeholder where Self: View`, only those ones declared as `extension MyView: Placeholder {}` could be used as a placeholder. 

I would prefer to keep it minimal, and let users to specify the conformance only when necessary.

Reply:
Check. That makes sense!